README.md

```diff
@@ -1,20 +1,193 @@
+![example workflow](https://github.com/marijnvanh/ft_IRC/actions/workflows/ci.yml/badge.svg)
 # ft_IRC
-The goal of this project is to make our own IRC server.
+As part of the Codam Advanced curriculum, students are expected to create a simple IRC server that handles basic client to client communication. The server needs to work with real IRC clients. For this server the irssi and kiwiirc clients were used. Multiple servers can be linked to each other to create an IRC network.
+
+# Table of content
+- [ft_IRC](#ft_irc)
+- [Table of content](#table-of-content)
+- [Requirements](#requirements)
+- [Quick start](#quick-start)
+  - [Local](#local)
+  - [Docker](#docker)
+- [Manual build](#manual-build)
+- [Running the server](#running-the-server)
+- [Config](#config)
+- [TLS](#tls)
+- [Testing methods](#testing-methods)
+  - [Unit testing](#unit-testing)
+  - [End to End tests](#end-to-end-tests)
+- [Supported client commands](#supported-client-commands)
+- [Shared Lib](#shared-lib)
+- [To be improved](#to-be-improved)
+- [Contributors](#contributors)
+- [Resources](#resources)
+  - [IRC Documentation](#irc-documentation)
+  - [Network programming](#network-programming)
+  - [OpenSSL](#openssl)
+  - [JSON parser](#json-parser)
+  - [Other recourses](#other-recourses)
+
 
 # Requirements
```

Comment:
For requirements, maybe good to note it doesn't inherently build with Musl libc?